tests/unit/test_dimension.py

```diff
@@ -724,6 +724,76 @@ def and_it_overrides_with_subtotal_transforms_when_present(
         _Subtotals_.assert_called_once_with(["subtotal", "dicts"], valid_elements_)
         assert subtotals is subtotals_
 
+    @pytest.mark.parametrize(
+        "dim_type, view_insertion_dicts, dimension_transforms_dict, insertion_dicts",
+        (
+            (DT.MR, [], {}, []),
+            (DT.CA_SUBVAR, [], {}, []),
+            (
+                DT.CAT,
+                [{"anchor": "top", "name": "a"}],
+                {"insertions": [{"anchor": "3", "name": "a", "hide": True}]},
+                [{"anchor": "top", "name": "a", "hide": True}],
+            ),
+            (
+                DT.CAT,
+                [
+                    {"anchor": "top", "name": "a"},
+                    {"anchor": "bottom", "name": "b", "hide": True},
+                ],
+                {
+                    "insertions": [
+                        {"anchor": "3", "name": "a", "hide": True},
+                        {"anchor": "2", "name": "b", "hide": True},
+                    ]
+                },
+                [
+                    {"anchor": "top", "name": "a", "hide": True},
+                    {"anchor": "bottom", "name": "b", "hide": True},
+                ],
+            ),
+            (
+                DT.CAT,
+                [{"anchor": "top", "name": "a"}],
+                {},
+                [{"anchor": "top", "name": "a"}],
+            ),
+        ),
```

Comment:
I think your code fails in the case of two insertions anchored to the same thing but with a flipped order between variable view & analysis. This is important because it their order breaks the tie when they are anchored to the same thing.

Here's another test case, can you add?
```python
(
  DT.CAT,
  [{"anchor": 1, "name": "a"}, {"anchor": 1, "name": "b"}],
  [{"anchor": 1, "name": "b"}, {"anchor": 1, "name": "a"}],
  [{"anchor": 1, "name": "a"}, {"anchor": 1, "name": "b"}],
)
```